integrations/ultralytics/deepsparse/client.py

```diff
@@ -43,21 +43,19 @@ def detect(
     ) -> List[numpy.ndarray]:
         """
         :param images: list of or singular file paths or numpy arrays of images to
-            run the detection model on
+            run the detection model on. Number of images should equal model batch size
         :return: list of post-processed object detection results for each image including
             class label, likelihood, and bounding box coordinates
         """
         if not isinstance(images, List):
             images = [images]
-        images = [
-            cv2.imread(image) if isinstance(image, str) else image for image in images
-        ]
+        images = numpy.stack([load_image(image) for image in images])
 
```

Comment:
does `len` work on numpy arrays?

Reply:
yes, where does `len` factor in here?

Reply:
sorry, i made this on the line below and it selected this instead
```
print(f"Sending batch of {len(images)} images for detection to {self._url}")
```

Reply:
ah, thanks.  images is a list at this point so should be fine.